File: Swift学习/code4xcode6/ch21/21.5Calculator(ObjC)/Calculator/ViewController.m
//
//  ViewController.m
//  Calculator
//
//  Created by tonymacmini on 14-9-20.
//  Copyright (c) 2014年 tonymacmini. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.mainLabel.text = @"0";
    logic = [[CalcLogic alloc] init];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)operandPressed:(id)sender {
    UIButton* btn = (UIButton*)sender;
    self.mainLabel.text =  [logic calculateByTag:(int)btn.tag withMainLabelString:self.mainLabel.text];
}

- (IBAction)equalsPressed:(id)sender {
    UIButton* btn = (UIButton*)sender;
    self.mainLabel.text =  [logic calculateByTag:(int)btn.tag withMainLabelString:self.mainLabel.text];
}

- (IBAction)clearPressed:(id)sender {
    self.mainLabel.text = @"0";
    [logic clear];
}

- (IBAction)decimalPressed:(id)sender {
    if ( [logic doesStringContainDecimal:self.mainLabel.text] == FALSE) {
        NSString *s =  [self.mainLabel.text stringByAppendingString:@"."];
        self.mainLabel.text  =  s;
    }
}

- (IBAction)numberButtonPressed:(id)sender {
    UIButton* btn = (UIButton*)sender;
    self.mainLabel.text = [logic updateMainLabelStringByNumberTag:(int)btn.tag
                                              withMainLabelString:self.mainLabel.text];
    
}

@end

File: Swift学习/code4xcode6/ch21/21.2~21.4Calculator(Swfit)/Calculator/ViewController.swift
//
//  ViewController.swift
//  Calculator
//
//  Created by tonymacmini on 14-9-20.
//  Copyright (c) 2014年 tonymacmini. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var mainLabel: UILabel!
    
    var logic : CalcLogic!

    override func viewDidLoad() {
        super.viewDidLoad()
        mainLabel.text = "0"
        logic = CalcLogic()
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        logic = nil
    }
    
    @IBAction func operandPressed(sender: AnyObject) {
        var btn : UIButton = sender as UIButton
        mainLabel.text =  logic.calculateByTag(btn.tag, withMainLabelString: mainLabel.text!)
    }
    
    @IBAction func equalsPressed(sender: AnyObject) {
        var btn : UIButton = sender as UIButton
        mainLabel.text =  logic.calculateByTag(btn.tag, withMainLabelString: mainLabel.text!)
    }
    
    @IBAction func clearPressed(sender: AnyObject) {
        mainLabel.text = "0"
        logic.clear()
    }

    @IBAction func decimalPressed(sender: AnyObject) {
        if logic.doesStringContainDecimal(mainLabel.text!) == false {
            let string = mainLabel.text! + "."
            mainLabel.text = string
        }
    }

    @IBAction func numberButtonPressed(sender: AnyObject) {
        var btn : UIButton = sender as UIButton
        mainLabel.text =  logic.updateMainLabelStringByNumberTag(btn.tag, 
                                    withMainLabelString: mainLabel.text!)
    }
}

